### DISCUSSION TITLE
tutorials/lift-off-part3/09-navigating-to-the-track-page

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#46] [course feedback]
# tutorials/lift-off-part3/09-navigating-to-the-track-page

https://www.apollographql.com/tutorials/lift-off-part3/09-navigating-to-the-track-page

**Comment by shakeelmazhar** [comment]
server cache is not working for me. It takes time to load data every time.
This is the code that I'm using.

const { url } = await startStandaloneServer(server, {
    context: async () => {
      const { cache } = server;

      return {
        dataSources: {
          trackAPI: new TrackAPI({ cache }),
        },
      };
    },
  });

**Reply by mabuyo** [reply]
Hey @shakeelmazhar , sorry we missed your comment! That code looks correct to me. How long were your queries taking? Maybe it was something on the RESTDataSource end that was configured incorrectly?

**Reply by AnsBdran** [reply]
you need to set the cache on the client side.
```javascript
const client = new ApolloClient({
    uri: 'http://localhost:4000',
    cache: new InMemoryCache()
})
```

**Comment by trogancodes** [comment]
Hey @mabuyo, please i am getting "Failed to load resource: the server responded with a status of 500 (Internal Server Error)" on the client side of the app, and apparently my track fields got erased on the server and i dont know how or why. the error is as a result of this problem

**Reply by mabuyo** [reply]
Hey there @trogancodes - my apologies for the delayed response, our notifications integration stopped working and I completely missed this. Are you still having issues with this? Happy to help troubleshoot although I realize it's been a while.

**Comment by ghost** [comment]
I was struggling a little bit with the styled in typescript in `const CardContainer = styled.div` and I solve it using useNavigate  
```js
 const { title, thumbnail, author, length, modulesCount, id } = track;
  const navigate = useNavigate();

  return (
    <CardContainer onClick={() => { navigate(`/track/${id}`) }}>
```

**Reply by austinav** [reply]
Yeah, I am not familiar with emotion and it wasn't finding the link to prop, so I left it a div and did what you did

**Comment by EdiFirst** [comment]
Hi team, excellent course! But I'm not sure why I'm getting issues on this last part, I finished part 3 answering correctly the question because I consume the values directly from Apollo studio. Let me explain, the initial page are loading normally, but when I click in any card , the url is formed like: http://localhost:3000/track/c_1, but I'm getting this error
ERROR: Response not successful: Received status code 400:
I took a look on console log: Failed to load resource: the server responded with a status of 400 (Bad Request), and there is a link to http://localhost:4000:1, when I click up on this its load the Apollo studio with the correct querie ( the same trackId of the card I clicked).

I double checked my code and review the module it's ok, I compared it with the folder final and it is same code. 
As I understood the concept and everything else, I continued on to part 4, after clone the repos and run the codes ( server and client ), It's works normally. I realized that index.js on src/page is different from part 3, here was add the modules:

```
import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
/** importing our pages */
import Tracks from './tracks';
import Track from './track';
import Module from './module';

export default function Pages() {
  return (
    <BrowserRouter>
      <Routes>
        <Route element={<Tracks />} path="/" />
        <Route element={<Track />} path="/track/:trackId" />
        <Route element={<Module />} path="/track/:trackId/module/:moduleId" />
      </Routes>
    </BrowserRouter>
  );
}
```

Although It make sense, I don't remember that was covered on part 3. Am I correct?  
Thank you in advance, great job Apollo team!

**Reply by lizhennessy** [reply]
Hi there @EdiFirst , you're absolutely right - the modules feature was added in the [Lift-off Lab](https://www.apollographql.com/tutorials/lift-off-lab) which fits between Lift-off III and IV. I am curious if you have your code pushed up to a repository I could take a look at? that error sounds strange and I'd love to check it out.

**Reply by EdiFirst** [reply]
Hi thank you @lizhennessy  I finish the course so I don't have my code in repos.

**Comment by piyushyadav1617** [comment]
The REST API does not seem to work most of the time it throws error: ERROR: request to https://odyssey-lift-off-rest-api.herokuapp.com/author/cat-1 failed, reason: getaddrinfo EAI_AGAIN odyssey-lift-off-rest-api.herokuapp.com
This happens both on the / and and /track/:trackId routes. It sometimes work but most of the time it throws this error.

**Reply by lizhennessy** [reply]
Thanks for the feedback @piyushyadav1617 ! We'll look into what's going on with the REST API. When I check out the origin of `getaddrinfo EAI_AGAIN` it seems like it might be a network connectivity issue. Are you perhaps using a work computer that might be having trouble accessing the address?

**Reply by piyushyadav1617** [reply]
> Thanks for the feedback @piyushyadav1617 ! We'll look into what's going on with the REST API. When I check out the origin of `getaddrinfo EAI_AGAIN` it seems like it might be a network connectivity issue. Are you perhaps using a work computer that might be having trouble accessing the address?

I am on my personal laptop. The issue is occurring only on my app and the explorer, but the fetch requests through swagger is running fine.

**Comment by agostinaietri** [comment]
Hi guys, I'm getting ERROR: Cannot return null for non-nullable field Module.id. when selecting a specific track. Could you point me into the right direction?

**Reply by mabuyo** [reply]
Hey @agostinaietri it sounds like there might be something wrong with the Track.modules resolver and it's not returning data for that particular track. Can you double check that section? Were you able to run the query through Sandbox successfully?

**Comment by virkumpavat** [comment]
Great Tutorial , Still able to make it without any issue
